src/api/firebase.js

```diff
@@ -221,3 +226,107 @@ export async function deleteItem(listPath, itemId) {
 		throw new Error(`Failed updating item: ${error.message}`);
 	}
 }
+
+export let urgencyObject = {
+	overdue: new Set(),
+	soon: new Set(),
+	kindOfSoon: new Set(),
+	notSoon: new Set(),
+	inactive: new Set(),
+};
+
+/**
+ * Sorts an item into one of four urgency categories based on the provided urgency status.
+ *
+ * Items are categorized into 'soon', 'kindOfSoon', 'notSoon', or 'inactive' based on the urgency.
+ *
+ * @param {Object} item - The item to be categorized. Should include at least a `name` property.
+ * @param {number} urgencyStatus - The urgency status of the item, used to determine its category.
+ * @throws Will throw an error if the item cannot be placed in a category.
+ */
+const sortByUrgency = (item, daysUntilNextPurchase) => {
+	if (item.name.includes('sun')) {
+		console.log(`${item.name} urgencyStatus ${daysUntilNextPurchase}`);
+	}
+	if (daysUntilNextPurchase < 0) {
+		urgencyObject.overdue.add(item);
+		return;
+	} else if (daysUntilNextPurchase === 1000) {
+		urgencyObject.inactive.add(item);
+		return;
+	} else if (daysUntilNextPurchase < 7) {
+		urgencyObject.soon.add(item);
+		return;
+	} else if (daysUntilNextPurchase >= 7 && daysUntilNextPurchase < 30) {
+		urgencyObject.kindOfSoon.add(item);
+		return;
+	} else if (daysUntilNextPurchase >= 30) {
+		urgencyObject.notSoon.add(item);
+		return;
+	} else {
+		throw new Error(`Failed to place [${item.name}]`);
+	}
+};
+
+/**
+ * Calculates the urgency of an item based on the number of days
+ * since it was last purchased or created, and the time until the next purchase date.
+ *
+ * @param {Object} item - The item object containing relevant date information.
+ * @param {string | null} item.dateLastPurchased - The date when the item was last purchased, or null if never purchased.
+ * @param {string} item.dateCreated - The date when the item was created.
+ * @returns {number} - The urgency of the item:
+ *  - `100` if more than 60 days have passed since the last purchase or creation,
+ *  - or the number of days until the next purchase if fewer than 60 days have passed.
+ */
+const getItemUrgency = (item) => {
+	// get date the item was purchased or created
+	const itemDate = getDateLastPurchasedOrDateCreated(
+		item.dateLastPurchased,
+		item.dateCreated,
+	);
+	// check how many days have passed since that date
+	const daysToToday = getDaysFromDate(itemDate);
+
+	// if more than 60 days have passed
+	if (daysToToday >= 60) {
+		// sort as inactive
+		sortByUrgency(item, 1000);
+		return 1000;
+	} else {
+		// sort by the amount of days until next purchase date
+		const daysUntilNextPurchase = getDaysBetweenDates(
+			new Date(),
+			item.dateNextPurchased.toDate(),
+		);
+		if (item.name.includes('sun')) {
+			console.log(
+				`${item.name} days until next purchase is ${daysUntilNextPurchase}`,
+			);
+		}
+		sortByUrgency(item, daysUntilNextPurchase);
+
+		return daysUntilNextPurchase;
+	}
+};
```

Comment:
This function appears to be purely client-side and doesn't interact with Firebase. Could we move it to a utility file and add a test to ensure its logic is covered?